beacon_node/rpc/src/beacon_node.rs

```diff
@@ -6,12 +6,21 @@ use protos::services_grpc::BeaconNodeService;
 use slog::{trace, warn};
 use std::sync::Arc;
 
-#[derive(Clone)]
 pub struct BeaconNodeServiceInstance<T: BeaconChainTypes> {
     pub chain: Arc<BeaconChain<T>>,
     pub log: slog::Logger,
 }
 
+// NOTE: Deriving Clone puts bogus bounds on T, so we implement it manually.
+impl<T: BeaconChainTypes> Clone for BeaconNodeServiceInstance<T> {
```

Comment:
On second thoughts, could probably get rid of this hack and just require `BeaconChainTypes: Clone`

Reply:
Tried it, and it was about as ugly. Still had to manually implement `Clone` in a few places, but they were more central, and I'd prefer to keep this hack at the edge.